lib/shim/message-shim/subscribe-consume.js

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const ATTR_DESTS = require('../../config/attribute-filter').DESTINATIONS
+const messageTransactionRecorder = require('../../metrics/recorders/message-transaction')
+const props = require('../../util/properties')
+const specs = require('../specs')
+
+/**
+ * Derives the transaction name based on the spec passed in
+ *
+ * @private
+ * @param {MessageShim} shim instance of shim
+ * @param {specs.MessageSubscribeSpec} msgDesc spec for function
+ * @returns {string} constructed name for transaction
+ */
+function _nameMessageTransaction(shim, msgDesc) {
+  let name = shim._metrics.LIBRARY + '/' + (msgDesc.destinationType || shim.EXCHANGE) + '/'
+
+  if (msgDesc.destinationName) {
+    name += shim._metrics.NAMED + msgDesc.destinationName
+  } else {
+    name += shim._metrics.TEMP
+  }
+
+  return name
+}
+
+/**
+ * Wrapper for subscribing to a queue.
+ *
+ * @param {object} params to function
+ * @param {MessageShim} params.shim instance of shim
+ * @param {Function} params.fn subscriber function
+ * @param {specs.MessageSubscribeSpec} params.spec spec for subscriber
+ * @param {boolean} params.destNameIsArg flag to state if destination is an argument
+ * @returns {Function} wrapped subscribe function
+ */
+module.exports = function createSubscriberWrapper({ shim, fn, spec, destNameIsArg }) {
```

Comment:
nit: I would refactor this to `module.exports = createSubscriberWrapper` at the top of the file. It makes it easier to see what the export is, where as the current declaration is easy to overlook.